mlrun/api/utils/projects/follower.py

```diff
@@ -88,6 +88,22 @@ def create_project(
             )
             created_project = None
             if not is_running_in_background:
+                # as part of the store_project flow we encountered an error related to the isolation level we use.
```

Comment:
I suggest to add a `commit_before_get` bool argument that will be `False` by default and will be set to `True` only when the `create_project` is called from `store_project`
In another words, this fix is hacky, so let's have it to happen only in the super specific flow where needed